service/create/service.go

```diff
@@ -73,34 +73,47 @@ func DefaultConfig() Config {
 		CertWatcher: nil,
 		K8sClient:   nil,
 		Logger:      nil,
-		S3Bucket:    "",
-		PubKeyFile:  "",
+
+		// Settings.
+		AwsConfig:  awsutil.Config{},
+		PubKeyFile: "",
 	}
 }
 
 // New creates a new configured service.
 func New(config Config) (*Service, error) {
 	// Dependencies.
-	// TODO other deps have to be checked as well
 	if config.CertWatcher == nil {
 		return nil, microerror.MaskAnyf(invalidConfigError, "config.CertWatcher must not be empty")
 	}
+	if config.K8sClient == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.K8sClient must not be empty")
+	}
 	if config.Logger == nil {
 		return nil, microerror.MaskAnyf(invalidConfigError, "config.Logger must not be empty")
 	}
 
+	// Settings.
+	var emptyAwsConfig awsutil.Config
+	if config.AwsConfig == emptyAwsConfig {
```

Comment:
That's pretty weird. This [works](https://play.golang.org/p/e7FxEefnLj) anyway:

    if (config.AwsConfig == awsutil.Config{})